playwright-gh-pages/deploy-report-pages/build-pr-comment.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const path = require('path');
+
+async function checkUrlExists(url) {
+  try {
+    const response = await fetch(url);
+    return response.status !== 404;
+  } catch (error) {
+    console.error('Error fetching URL:', error);
+    return false; // Assume false if there's an error (e.g., network issue)
+  }
+}
+
+async function buildPrComment() {
+  // Ensure we are in the right directory
+  const reportsDir = 'all-reports';
+  if (!fs.existsSync(reportsDir) || !fs.statSync(reportsDir).isDirectory()) {
+    console.error('Failed to enter directory all-reports');
+    process.exit(1);
+  }
+
+  // Initialize the table variable
+  let table = '### Playwright test results';
+
+  // Check if any summary.txt has a PLUGIN_NAME value
+  const summaryFiles = fs
+    .readdirSync(reportsDir)
+    .map((dir) => path.join(reportsDir, dir, 'summary.txt'))
```

Comment:
`dir` can be a file too here so you might be building incorrect paths. YOu need to check `dir` is a dir before returning it.